### PR TITLE
Capture stderr when execing credential provider fails

### PR DESCRIPTION
**Pull request by dims**
When we execute credential provider plugins, we just get a thumbs up and thumbs down in the return exit code currently. 

We should be able to at least log something printed into stderr when credential provider fails!

This scenario came up in conversations during https://github.com/kubernetes/cloud-provider-aws/pull/603

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by dims** [comment]
cc @mmerkes

File: pkg/credentialprovider/plugin/plugin.go [hunk 1]
```diff
@@ -445,8 +445,13 @@ func (e *execPlugin) runPlugin(ctx context.Context, cmd *exec.Cmd, image string)
 		return fmt.Errorf("error execing credential provider plugin %s for image %s: %w", e.name, image, ctx.Err())
 	}
 	if err != nil {
+		stderr := ""
+		if err2, ok := err.(*exec.ExitError); ok {
```
Comment (aojea) [inline pkg/credentialprovider/plugin/plugin.go:449]: naive question, is there any possibility of leaking sensitive information?

Comment (mmerkes) [reply pkg/credentialprovider/plugin/plugin.go:449]: That's a fair question. However, if the credential providers are logging sensitive information, that's where the bug is. They shouldn't be logging creds or anything.

Comment (aojea) [reply pkg/credentialprovider/plugin/plugin.go:449]: fair enough

**Comment by mmerkes** [comment]
@dims Looks like @cartermckinnon made a similarly intended change with #117448. The result will more or less achieve that same thing, right? Looks like they'll play nicely together though.

**Comment by dims** [comment]
@mmerkes if just the already merged patch from @cartermckinnon works then we don't need this!

/hold

**Comment by dims** [comment]
@mmerkes @cartermckinnon so do try out w/o this PR. if it works, i'll close this out. I think information about the stderr may be lost here:

https://github.com/kubernetes/kubernetes/pull/117448/files#diff-ea86580504db5fcaa2dcc603216cbd2ecafaa20f897fff75bb73bf06598f234dR442-R446

**Comment by cartermckinnon** [comment]
@dims the `cmd` we pass to `runPlugin` has its `Stderr` set here (in the calling function): https://github.com/cartermckinnon/kubernetes/blob/766f6c3ad4133df7bf096fbed61c24bcd19c25c8/pkg/credentialprovider/plugin/plugin.go#L399

So if `runPlugin` returns an error, we're still able to access what was written to the `stderr` buffer.

That being said -- I'll do another sanity check. 😄

**Comment by mmerkes** [comment]
@dims @cartermckinnon I setup a 1.26 EKS cluster, built and installed a custom `ecr-credential-provider` with [additional logging that won't be necessary](https://github.com/kubernetes/cloud-provider-aws/compare/master...mmerkes:cloud-provider-aws:ecr-logging) because we already log the error [here](https://github.com/kubernetes/cloud-provider-aws/blob/master/cmd/ecr-credential-provider/main.go#L243), built and install the latest kubelet, and updated the image credential provider config to authenticate against public ECR without giving proper permissions.

Long story short, it works! See the output below:

```
May  9 22:33:22 ip-192-168-10-7 kubelet: E0509 22:33:22.891641    4189 plugin.go:233] Failed getting credential from external registry credential provider: error execing credential provider plugin ecr-credential-provider for image public.ecr.aws/nginx/nginx: exit status 1: I0509 22
:33:22.572583    6682 main.go:93] Getting creds for public registry
May  9 22:33:22 ip-192-168-10-7 kubelet: E0509 22:33:22.890550    6682 plugin.go:68] Error running credential provider plugin: AccessDeniedException: User: arn:aws:sts::65013231XXXX:assumed-role/eksctl-metrics-nodegroup-ng-db4fa-NodeInstanceRole-DASJ5HCXMK80/i-006dba522499a1a37 is
not authorized to perform: ecr-public:GetAuthorizationToken on resource: * because no identity-based policy allows the ecr-public:GetAuthorizationToken action
May  9 22:33:22 ip-192-168-10-7 kubelet: status code: 400, request id: b00eade3-a60f-4085-9fb5-06cb3dad3d10
May  9 22:33:22 ip-192-168-10-7 kubelet: E0509 22:33:22.890575    6682 main.go:243] Error running credential provider plugin: AccessDeniedException: User: arn:aws:sts::65013231XXXX:assumed-role/eksctl-metrics-nodegroup-ng-db4fa-NodeInstanceRole-DASJ5HCXMK80/i-006dba522499a1a37 is n
ot authorized to perform: ecr-public:GetAuthorizationToken on resource: * because no identity-based policy allows the ecr-public:GetAuthorizationToken action
May  9 22:33:22 ip-192-168-10-7 kubelet: status code: 400, request id: b00eade3-a60f-4085-9fb5-06cb3dad3d10
```

I don't think this PR is necessary, though I appreciate the fact that you wrote unit tests for this.

**Comment by dims** [comment]
thanks @cartermckinnon !

**Pull request closed without merging**